src/typescript-service.ts

```diff
@@ -1267,7 +1267,7 @@ export class TypeScriptService {
 		// Ensure files needed for most operations are fetched
 		await this.projectManager.ensureReferencedFiles(uri).toPromise();
 		this.projectManager.didOpen(uri, params.textDocument.text);
-		this._publishDiagnostics(uri);
+		setTimeout(() => this._publishDiagnostics(uri), 200);
```

Comment:
if `publishDiagnostics` throws this crashes the process. I would do this:
```ts
await new Promise(resolve => setTimeout(resolve, 200))
this._publishDiagnostics(uri)
```

Reply:
Whoops, all error handling out the window.
Cleaned up the tests after applying your suggestion, thanks for the help!